.buildkite/baristacli.py

```diff
@@ -0,0 +1,569 @@
+
+# default path of configuration file
+CONFIGFILE = os.path.join(os.path.basename(os.path.dirname(__file__)),
+                          "baristaPipeline.yml")
+
+# default bazel binary
+BAZELBIN = "bazel"
+
+# configuration of allowed buildkite instances
+# PLATFORMS = {
+#   "default" : {            # BuildKite queue name
+#     "name": "AWS Linux 2", # label
+#     "platform": "linux"    # specify platform (linux/windows)
+#   },
+# }
+PLATFORMS = {
+    "default": {
+        "name": ":linux: AWS Linux 2",
+        "platform": "linux"
+    },
+    "windows10": {
+        "name": ":windows: Windows 10",
+        "platform": "windows"
+    }
+}
+
+class BuildKiteConfigError(ValueError):
+    """ BuildKiteConfigError Exception """
+
+class BazelFailingTests(Exception):
+    """ BazelFailingTests Exception"""
+
+
+def is_windows():
+    """
+    Return True if we are on windows.
+    """
+    return os.name == "nt"
+
+
+# set script extension depending on platform
+SCRIPT_EXTENSION = ".sh" if not is_windows() else ".bat"
+
+
+def eprint(*args, **kwargs):
+    """
+    Print to stderr and flush (just in case).
+    """
+    print(*args, flush=True, file=sys.stderr, **kwargs)
+
+
+def execute_shell_commands(commands):
+    """
+    Prepare given list of commands and execute it
+    :param commands: array if commands
+    """
+    if not commands:
+        return
+    if is_windows():
+        print_collapsed_group(":cmd: Setup (Windows Commands)")
+        for cmd in commands:
+            execute_command(cmd.split(" "), shell=True)
+    else:
+        print_collapsed_group(":bash: Setup (Shell Commands)")
+        shell_command = "\n".join(["set -e"] + commands)
+        execute_command([shell_command], shell=True)
+
+
+def execute_command(args, shell=False, fail_if_nonzero=True, cwd=None,
+                    print_output=True):
+    """
+    Execute command
+    """
+    if print_output:
+        eprint(" ".join(args))
+    return subprocess.run(
+        args, shell=shell, check=fail_if_nonzero, env=os.environ, cwd=cwd
+    ).returncode
+
+
+def arg_hander_command(args):
+    """
+    Prepare the given platform and task to be able to call it
+    """
+    tasks = load_config(args.config)
+
+    if not args.platform in tasks["tasks"].keys():
+        raise BuildKiteConfigError(
+            "{} is not available. Please specify one of {}".format(args.platform,
+                                                                   PLATFORMS.keys()))
+
+    stage = tasks["tasks"].get(args.platform)
+
+    if not args.target in stage:
+        raise BuildKiteConfigError(
+            "{} is not available. Please specify one of {}".format(args.target,
+                                                                   stage.keys()))
+
+    # execute pre commands
+    if "pre_cmd" in stage[args.target]:
+        execute_shell_commands(stage[args.target]['pre_cmd'])
+
+    # bazel build info file
+    build_event_file = "build_event_file.json"
```

Comment:
for what do we need the build event prrotocol?

Reply:
publish tests in buildkite if they fail